src/Library/Plugin/Notice.php

```diff
@@ -0,0 +1,295 @@
+<?php
+/**
+ * BoldGrid Library Plugin Page Notice.
+ *
+ * @package Boldgrid\Plugin
+ *
+ * @since SINCEVERSION
+ * @author BoldGrid <wpb@boldgrid.com>
+ */
+
+namespace Boldgrid\Library\Library\Plugin;
+
+/**
+ * Notice class for Plugin\Page.
+ *
+ * This class is a specific Notice
+ * used by the Boldgrid\Library\Library\Plugin\Page 
+ * and Boldgrid\Library\Library\Plugin\Plugin classes.
+ *
+ * @since SINCEVERSION
+ */
+
+class Notice {
+
+    /**
+     * Notice ID
+     * 
+     * @var string
+     * @access protected
+     * @since SINCEVERSION
+     */
+    protected $id;
+
+    /**
+     * Notice Page slug
+     * 
+     * @var string
+     * @access protected
+     * @since SINCEVERSION
+     */
+    protected $pageSlug;
+
+    /**
+     * Notice Version
+     * 
+     * Version of plugin this notice was added on
+     * 
+     * @var string
+     * @access protected
+     * @since SINCEVERSION
+     */
+    protected $version;
+
+    /**
+     * Plugin Object
+     * 
+     * Plugin Object this belongs to
+     * 
+     * @var Plugin
+     * @access protected
+     * @since SINCEVERSION
+     */
+    protected $plugin;
+
+    /**
+     * Notice Is Unread
+     * 
+     * Specifies if the Notice is Unread or not
+     * 
+     * @var bool
+     * @access protected
+     * @since SINCEVERSION
+     */
+    protected $isUnread;
+
+    /**
+     * Constructor 
+     * 
+     * @param Plugin $plugin Plugin instance that this Notice belongs to
+     * @param array $notice An array of Notice data from plugin config
+     * @since SINCEVERSION
+     * 
+     */
+    public function __construct( Plugin $plugin, array $notice ) {
+        if ( false !== $this->alreadyExists( $notice['id'] ) ) {
```

Comment:
What do you think of changing this line for readability?

`false !== $this->alreadyExists( $notice['id'] )`

As it is now, I read it as:
`if false does not equal alreadyExists()`
... it takes me a minute to understand what that means.

But if we change it to either of the following, it makes it easier to understand:

`if ( ! alreadyExists() )`
if it does not already exist

`if ( alreadyExists() )`
if it already exists

Reply:
This test has to be strictly test whether or not alreadyExists returns a false boolean value. To improve readability, I changed the code as follows:

`if ( false === $this->alreadyExists( $notice['id'] ) ) {`
`Notice Does Not Exist`
`} else {`
`Notice Does Exist`
`}`